Xamarin.Essentials/DeviceDisplay/DeviceDisplay.shared.cs

```diff
@@ -35,7 +46,13 @@ static void OnScreenMetricsChanged(ScreenMetrics metrics)
             => OnScreenMetricsChanged(new ScreenMetricsChangedEventArgs(metrics));
 
         static void OnScreenMetricsChanged(ScreenMetricsChangedEventArgs e)
-            => ScreenMetricsChangedInternal?.Invoke(null, e);
+        {
+            if (!currentMetrics.Equals(e.Metrics))
+            {
+                SetCurrent();
```

Comment:
Maybe rather than getting the current again, just pass `e.Metrics` to an overload or as a parameter (can be a nullable as it is internal) Some platforms may be expensive later on

Reply:
Makes sense!